eng/pipelines/templates/build-pull-request.yml

```diff
@@ -41,13 +47,27 @@ jobs:
 
   # Publishes the test results to the Azure Pipeline itself so they can be viewed in the UI.
   # This needs to be ran after the build, because if the build failed due to a test failure, the test results wouldn't be published.
+  # This uploads for .NET Framework run on main branch
   - task: PublishTestResults@2
-    displayName: Publish Test Results
+    displayName: Publish Test Results (XUnit)
     inputs:
       testRunner: XUnit
       testResultsFiles: $(Build.SourcesDirectory)/artifacts/$(BuildConfiguration)/TestResults/*.xml
       testRunTitle: Unit Test Results
     condition: succeededOrFailed()
+    continueOnError: true
+
+  # Publishes the test results to the Azure Pipeline itself so they can be viewed in the UI.
+  # This needs to be ran after the build, because if the build failed due to a test failure, the test results wouldn't be published.
+  # This uploads for .NET Core run on C# dev kit branch
```

Comment:
Since C# dev kit and main branch runs tests different way (xunit.exe vs dotnet test) it produces different artifacts. Currently we just try to copy artifacts both ways in PR. (one of them will pass, other will fail depending on way to run).
The very next thing I am planning to work is combining testing mechanisms between both branches.